### PR TITLE
Gidk 85 dev: ledger pagination and GST invoice download implemented

### PR DESCRIPTION
**Pull request by rohit-walkover**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

File: apps/web-giddh/src/app/invoice/generate/invoice.generate.component.scss [hunk 1]
```diff
@@ -181,16 +191,56 @@ button.btn.btn-filter {
...
+.pending-table-description {
     white-space: normal;
 }
+
+.close-modal img {
+    width: 16px;
+    opacity: 0.6;
+}
+
+.invoice-editWrapper {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/invoice/generate/invoice.generate.component.scss:204]: Avoid using improper class names

File: apps/web-giddh/src/app/invoice/generate/invoice.generate.component.scss [hunk 2]
```diff
@@ -181,16 +191,56 @@ button.btn.btn-filter {
...
+    height: 100vh;
+    overflow-y: auto;
+    .invoice-editHeader {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/invoice/generate/invoice.generate.component.scss:207]: Avoid using improper class names

File: apps/web-giddh/src/app/invoice/generate/invoice.generate.component.scss [hunk 3]
```diff
@@ -181,16 +191,56 @@ button.btn.btn-filter {
...
+            width: 32px;
+            background-color: #ddd;
+            border: 1px solid $gainsboro;
+            display: flex;
+            justify-content: center;
+            align-items: center;
+            margin-right: 10px;
+        }
+    }
+    .invoice-editBody {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/invoice/generate/invoice.generate.component.scss:233]: Avoid using improper class names

File: apps/web-giddh/src/app/invoice/generate/invoice.generate.component.ts [hunk 4]
```diff
@@ -222,21 +234,35 @@ export class InvoiceGenerateComponent implements OnInit, OnChanges, OnDestroy {
                 }
             });
 
-        this.store.select(p => p.receipt.voucher).pipe(
+        // this.store.select(p => p.receipt.voucher).pipe(
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/invoice/generate/invoice.generate.component.ts:237]: Can we remove commented code

File: apps/web-giddh/src/app/invoice/generate/invoice.generate.component.ts [hunk 5]
```diff
@@ -337,15 +364,15 @@ export class InvoiceGenerateComponent implements OnInit, OnChanges, OnDestroy {
...
-    public closeInvoiceModel(e) {
-        if (e.action === 'generate') {
-            this.selectedLedgerItems = [];
-        }
-        this.invoiceGenerateModel.hide();
-        setTimeout(() => {
-            this.store.dispatch(this.invoiceActions.ResetInvoiceData());
-        }, 2000);
-    }
+    // public closeInvoiceModel(e) {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/invoice/generate/invoice.generate.component.ts:367]: Can we remove commented code

File: apps/web-giddh/src/app/invoice/generate/invoice.generate.component.ts [hunk 6]
```diff
@@ -446,9 +490,9 @@ export class InvoiceGenerateComponent implements OnInit, OnChanges, OnDestroy {
         }
     }
 
-    public showInvoiceModal() {
-        this.invoiceGenerateModel.show();
-    }
+    // public showInvoiceModal() {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/invoice/generate/invoice.generate.component.ts:493]: Can we remove commented code

File: apps/web-giddh/src/app/invoice/generate/invoice.generate.component.ts [hunk 7]
```diff
@@ -654,9 +698,21 @@ export class InvoiceGenerateComponent implements OnInit, OnChanges, OnDestroy {
 
     public resetDateSearch() {
         this.ledgerSearchRequest.dateRange = this.universalDate;
+        if (!this.universalDate) {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/invoice/generate/invoice.generate.component.ts:701]: Consider removing empty if block

File: apps/web-giddh/src/app/invoice/generate/invoice.generate.component.ts [hunk 8]
```diff
@@ -654,9 +698,21 @@ export class InvoiceGenerateComponent implements OnInit, OnChanges, OnDestroy {
...
+
+        }
         this.datePickerOptions = { ...this.datePickerOptions, startDate: moment(this.universalDate[0], 'DD-MM-YYYY').toDate(), endDate: moment(this.universalDate[1], 'DD-MM-YYYY').toDate() };
         this.isUniversalDateApplicable = true;
         this.getLedgersOfInvoice();
         this.detectChanges();
     }
+
+       public toggleBodyClass() {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/invoice/generate/invoice.generate.component.ts:710]: Add JS doc and return type

File: apps/web-giddh/src/app/invoice/generate/invoice.generate.component.ts [hunk 9]
```diff
@@ -394,15 +421,32 @@ export class InvoiceGenerateComponent implements OnInit, OnChanges, OnDestroy {
...
-        this.showInvoiceModal();
+        // this.showInvoiceModal();
+    }
+
+    /**
+     * To close preview model
+     *
+     * @memberof InvoiceGenerateComponent
+     */
+    public closeModel() {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/invoice/generate/invoice.generate.component.ts:445]: Return type

File: apps/web-giddh/src/app/invoice/generate/invoice.generate.component.ts [hunk 10]
```diff
@@ -163,6 +165,14 @@ export class InvoiceGenerateComponent implements OnInit, OnChanges, OnDestroy {
     public isUniversalDateApplicable: boolean = false;
     private isBulkInvoiceGeneratedWithoutErr$: Observable<boolean>;
     public isMobileView = false;
+    /** To show edit mode component */
+    public showEditMode: boolean = false;
+    /** get voucher details response object*/
+    public voucherDetails: any;
+    public selectedItem: InvoicePreviewDetailsVm;
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/invoice/generate/invoice.generate.component.ts:172]: Missing JS docs of variables

File: apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts [hunk 11]
```diff
@@ -3558,7 +3564,7 @@ export class ProformaInvoiceComponent implements OnInit, OnDestroy, AfterViewIni
     }
 
     public ngOnDestroy() {
-        if (!this.isProformaInvoice && !this.isEstimateInvoice) {
+        if (!this.isProformaInvoice && !this.isEstimateInvoice && !this.isPendingVoucherType) {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts:3567]: Add same condition on router NavigationStart listener in same file

File: apps/web-giddh/src/app/invoice/generate/invoice.generate.component.ts [hunk 12]
```diff
@@ -256,7 +282,8 @@ export class InvoiceGenerateComponent implements OnInit, OnChanges, OnDestroy {
                 this.universalDate = _.cloneDeep(dateObj);
                 this.ledgerSearchRequest.dateRange = this.universalDate;
                 this.datePickerOptions = { ...this.datePickerOptions, startDate: moment(this.universalDate[0], 'DD-MM-YYYY').toDate(), endDate: moment(this.universalDate[1], 'DD-MM-YYYY').toDate(), chosenLabel: this.universalDate[2] };
-
+                this.fromDate = moment(this.universalDate[0], 'DD-MM-YYYY').toDate().toString();
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/invoice/generate/invoice.generate.component.ts:285]: Use constant for date

File: apps/web-giddh/src/app/invoice/generate/invoice.generate.component.ts (hunk 9 above)
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/invoice/generate/invoice.generate.component.ts:445]: add return type

File: apps/web-giddh/src/app/invoice/generate/invoice.generate.component.ts (hunk 8 above)
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/invoice/generate/invoice.generate.component.ts:710]: return type

**Pull request closed without merging**